docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+### Module: `./src/constants.tsx`
+
+```tsx
+export const DS_CONFIG = 'datasetConfig'
+export const MDL_CONFIG = 'modelConfig'
+export const EVAL_CONFIG = 'modelEvaluation'
+```
+
+Next, we need a directory to store our components. In addition, a subdirectory for our charting component.
+You should now have
+
+```txt
+|- src/
+  |- components/
+    |- charting/
+  |- index.html
+  |- index.tsx
+  ...
+  |- webpack.prod.js
+```
+
+### Module:  `./src/components/*`
+We can now add the following to our `components` directory.
+**Note**: We are again, but for the _frontend_ SDK, creating **reusable** components we could import or place into other tools we or others develop (_depending on a publisher's license and availability_) using your components and vice-versa!
+In other words, when we create our first file below, `config-inputs.tsx`, we are creating a component to sprinkle about different places and avoid copying, pasting, and in turn, propagating stale code causing all sorts of sneaky bugs in the near (or even scarier... **FAR** future!!) future.
+
+#### Component(s): `config-inputs.tsx`
+
+```jsx
+import React, { useContext } from 'react';
+import { TextField } from '@alteryx/ui';
+import { Context as UiSdkContext } from '@alteryx/react-comms';
+import _ from 'lodash'
+
+
+const ConfigTextInput = ({value, configType, vKey, elId, label, inputType="string"}) => {
+    const [model, handleUpdateModel] = useContext(UiSdkContext);
+
+    const onHandleTextChange = (event, configType, key) => {
+        const newModel = _.cloneDeep(model);
+        newModel.Configuration[configType][key] = event.target.value;
+        handleUpdateModel(newModel);
+    };
+
+    return (<TextField
+          fullWidth
+          type={inputType}
+          id={elId}
+          label={label}
+          onChange={(e) => onHandleTextChange(e, configType, vKey)}
+          value={value}
+        />)
+    }
+
+export {ConfigTextInput};  
+```
+
+**Important Note**: One of the few things we will explicitly call out here for the UI SDK, as it is so critical to developing.
+React's state model will _not_ "properly" update nested lists (as seen in our defaultConfig) or objects due to how it tracks changes.
+Therefore, we use `lodash` as the `alteryx-ui` docs recommended to ensure our `model` stays up to date without stale or redundant data.
+A common symptom of this issue is self-replicating lists, i.e.:
+
+```js
+// where model will = {someValue: [1], valueToUpdate: 5}
+const [model, handleUpdateModel] = useContext(UiSdkContext);
+// Create a shallow copy instead of using `lodash` or other method of deepcopy
+a = model;
+
+a.valueToUpdate = 8;
+// then use `handleUpdateModel` call
+// ... promise resovles and..
+>> 'a: {someValue: [1, 1], valueToUpdate: 8}'
+```
+
+Overall, this simple reusable component allows text input field values to be stored and updated via a given key and value within our config.
+You could add other inputs and fields like `FileInput` in this module.
```

Comment:
TW review:

Overall, this simple reusable component allows Text Input tool field values to be stored and updated via a given key and value within the config. You can add other inputs and fields like `FileInput` in this module.